workbench-core/accounts/src/services/projectService.test.ts

```diff
@@ -415,189 +500,808 @@ describe('ProjectService', () => {
+          return marshall(item);
+        }),
+        $metadata: {}
+      };
+      const user: AuthenticatedUser = {
+        id: 'user-123',
+        roles: ['ITAdmin']
+      };
+
+      // mock getUserGroups--TODO update after dynamic AuthZ intergration
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      jest.spyOn(ProjectService.prototype as any, '_mockGetUserGroups').mockImplementation(() => ['ITAdmin']);
+
       ddbMock
         .on(QueryCommand, {
           TableName: 'exampleDDBTable',
-          IndexName: 'getResourceByCreatedAt',
+          IndexName: 'getResourceByName',
           KeyConditionExpression: '#resourceType = :resourceType',
-          ExpressionAttributeNames: { '#resourceType': 'resourceType' },
+          ExpressionAttributeNames: {
+            '#resourceType': 'resourceType'
+          },
           ExpressionAttributeValues: {
-            ':resourceType': { S: 'project' },
-            ':proj1': { S: 'proj-123' },
-            ':proj2': { S: 'proj-123' },
-            ':proj3': { S: 'proj-123' },
-            ':proj4': { S: 'proj-123' }
+            ':resourceType': {
+              S: 'project'
+            }
           },
-          FilterExpression: 'id = :proj1 OR id = :proj2 OR id = :proj3 OR id = :proj4',
-          Limit: pageSize,
-          ExclusiveStartKey: marshall(lastEvaluatedKey)
+          Limit: 50,
+          ScanIndexForward: true
         })
-        .resolvesOnce(queryItemResponse2);
+        .resolves(queryItemResponse);
 
       // OPERATE
-      const actualResponse = await projService.listProjects({ user, pageSize });
+      const actualResponse = await projService.listProjects({
+        user,
+        sort: { name: 'asc' }
+      });
+
+      // CHECK
+      expect(actualResponse.data).toEqual([proj]);
+    });
+
+    test('list all projects as IT Admin on more than 1 page -- getting first page', async () => {
+      // BUILD
+      const items = [projItem, projItem];
+      const lastEvaluatedKey = {
+        pk: projItem.pk,
+        sk: projItem.sk,
+        resourceType: projItem.resourceType,
+        createdAt: projItem.createdAt
+      };
+      const paginationToken = Buffer.from(JSON.stringify(lastEvaluatedKey)).toString('base64');
+      const pageSize = 2;
+      const expectedResponse = { data: [proj, proj], paginationToken: paginationToken };
+      const queryItemResponse: QueryCommandOutput = {
+        Items: items.map((item) => {
+          return marshall(item);
+        }),
+        LastEvaluatedKey: marshall(lastEvaluatedKey),
+        $metadata: {}
+      };
+      const user: AuthenticatedUser = {
+        id: 'user-123',
+        roles: ['ITAdmin']
+      };
+
+      // mock getUserGroups--TODO update after dynamic AuthZ intergration
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      jest.spyOn(ProjectService.prototype as any, '_mockGetUserGroups').mockImplementation(() => ['ITAdmin']);
+
+      ddbMock
```

Comment:
In the future we should probably not mock the database and instead mock at the service layer. In this case we'd be mocking the `execute()` method on Query. That helps us decouple from the underlying library we use to connect to dynamo. As its currently written, we have to go and replace all the tests if we swap libraries even though the tests don't really care about what is being used to call dynamo.

Reply:
Tyler and I talked about this and I may come back and use this mocking method in the future.